src/main/java/com/aws/greengrass/secretmanager/SecretManager.java

```diff
@@ -328,10 +331,10 @@ public String validateSecretId(String secretId) throws GetSecretException {
         }
         // normalize name to arn
         if (!Pattern.matches(VALID_SECRET_ARN_PATTERN, secretId)) {
-            if (!nametoArnMap.containsKey(secretId)) {
+            if (!nameToArnMap.containsKey(secretId)) {
                 throw new GetSecretException(404, secretNotFoundErr + secretId);
             }
-            arn = nametoArnMap.get(secretId);
+            arn = nameToArnMap.get(secretId);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.



**Problem**  
 You are using a `ConcurrentHashMap`, but your usage of `containsKey()` and `get()` may not be thread-safe at lines: **334 and 337**. In between the check and the `get()` another thread can remove the key and the `get()` will return `null`. The remove that can remove the key is at line: **192**.

**Fix**  
 Consider calling `get()`, checking instead of your current check if the returned object is `null`, and then using that object only, without calling `get()` again.

**More info**  
 [View an example on GitHub](https://github.com/apache/hadoop/blob/f16cf877e565084c66bc63605659b157c4394dc8/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/s3guard/S3Guard.java#L302-L304) (external link).